c10/core/DispatchKeySet.h

```diff
@@ -195,102 +467,130 @@ class DispatchKeySet final {
-    DispatchKey::AutogradCPU,
-    DispatchKey::AutogradCUDA,
-    DispatchKey::AutogradXLA,
-    DispatchKey::AutogradLazy,
-    DispatchKey::AutogradNestedTensor,
-    DispatchKey::AutogradMLC,
-    DispatchKey::AutogradHPU,
-    DispatchKey::AutogradXPU,
-    DispatchKey::AutogradPrivateUse1,
-    DispatchKey::AutogradPrivateUse2,
-    DispatchKey::AutogradPrivateUse3,
-    DispatchKey::AutogradOther,
-});
+        DispatchKey::AutogradFunctionality,
+        DispatchKey::AutogradOther,
+        DispatchKey::AutogradNestedTensor,
+    });
 
 constexpr DispatchKeySet autocast_dispatch_keyset = DispatchKeySet({
-    DispatchKey::AutocastCPU,
-    DispatchKey::AutocastCUDA,
-});
+        DispatchKey::AutocastCPU,
+        DispatchKey::AutocastCUDA,
+    });
 
 // See Note [TLS Initialization]
 constexpr DispatchKeySet default_included_set = DispatchKeySet({
-    DispatchKey::BackendSelect,
-    DispatchKey::ADInplaceOrView,
-});
+        DispatchKey::BackendSelect,
+        DispatchKey::ADInplaceOrView,
+    });
 
 constexpr DispatchKeySet default_excluded_set = DispatchKeySet({
-    DispatchKey::AutocastCPU,
-    DispatchKey::AutocastCUDA,
-});
+        DispatchKey::AutocastCPU,
+        DispatchKey::AutocastCUDA,
+    });
 
 constexpr DispatchKeySet autograd_dispatch_keyset_with_ADInplaceOrView =
-    autograd_dispatch_keyset | DispatchKeySet(DispatchKey::ADInplaceOrView);
+        autograd_dispatch_keyset | DispatchKeySet(DispatchKey::ADInplaceOrView);
 
 // backend dispatch keys that map to DispatchKey::AutogradOther
 // NB: keys in this set also get associated with CompositeImplicitAutograd
 constexpr DispatchKeySet autogradother_backends = DispatchKeySet(
-    {DispatchKey::HIP,
-     DispatchKey::VE,
-     DispatchKey::FPGA,
-     DispatchKey::ORT,
-     DispatchKey::Vulkan,
-     DispatchKey::Metal,
-     DispatchKey::QuantizedCPU,
-     DispatchKey::QuantizedCUDA,
-     DispatchKey::CustomRNGKeyId,
-     DispatchKey::MkldnnCPU,
-     DispatchKey::SparseCPU,
-     DispatchKey::SparseCUDA,
-     DispatchKey::SparseHIP,
-     DispatchKey::SparseVE,
-     DispatchKey::SparseXPU,
-     DispatchKey::SparseCsrCPU,
-     DispatchKey::SparseCsrCUDA,
-     DispatchKey::Meta});
+        // HIP and VE aren't in this list: they now have their own backend bits
+        // which means that they can now have their own Autograd keys.
+        // Technically, HIP will now redispatch to its own custom AutogradHIP slot
+        // in the runtime table.
+        {DispatchKey::FPGA,
+        DispatchKey::ORT,
+        DispatchKey::Vulkan,
+        DispatchKey::Metal,
+        DispatchKey::SparseCsrCPU,
+        DispatchKey::SparseCsrCUDA,
+        DispatchKey::CustomRNGKeyId,
+        DispatchKey::MkldnnCPU,
+        DispatchKey::Meta,
+        // Sparse and Quantized backends also live here.
+        DispatchKey::Sparse,
+        DispatchKey::Quantized})
+        | DispatchKeySet(DispatchKeySet::RAW, full_backend_mask);
```

Comment:
Why add all the backend bits to this?

Reply:
`c10::autogradother_backends` is used during op registration ([here](https://github.com/pytorch/pytorch/blob/f37d2046f8c2d7d369919b0fb9d0ec0158634504/aten/src/ATen/core/dispatch/OperatorEntry.cpp#L270)), and that function needs to be able to loop over all of the runtime keys.

I've been thinking about it mostly as "any keysets used directly during registration probably need to include the backend bits". I'll add a comment about it.